content/en/blog/_posts/2021-11-23-object-storage-support.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: blog
+title: 'Introducing COSI: Object Storage Support in Kubernetes'
+date: 2021-11-23
+slug: cosi-object-storage-support
+---
+
+**Authors:** Sidhartha Mani (MinIO)
+
+This article introduces the Container Object Storage Interface (COSI), a standard for provisioning and consuming object storage in Kubernetes. It is a pre-alpha feature in Kubernetes v1.23.
+
+## Motivation
+
+File and block storage are treated as first class citizens in the Kubernetes ecosystem via CSI. Workloads using CSI volumes enjoy the benefits of portability across vendors and across Kubernetes clusters without the need to change application manifests. An equivalent standard does not exist for Object storage.
+
+Object storage has been rising in popularity in recent years as an alternative form of storage to filesystems and block devices. Object storage paradigm promotes disaggregation of compute and storage. This is done by making data available over the network, rather than locally. Disaggregated architectures allow compute workloads to be stateless, which consequently makes them easier to manage, scale and automate.
+
+## COSI
+ 
+COSI aims to standardize consumption of object storage to provide the following benefits:
+
+
+   * Kubernetes Native - A completely kubernetes-API driven experience to provision, configure and manage buckets
+   * Self Service - A clear delineation between administration and operations (DevOps) to enable self-service capability for DevOps personnel
+   * Portability - Vendor Neutrality enabled through portability across Kubernetes Clusters and across Object Storage vendors*
+
+_Portability across vendors is only possible when both vendors support a common datapath-API. Eg. it is possible to port from AWS S3 to Ceph, or AWS S3 to MinIO and back as they all use S3 API. In contrast, it is not possible to port from AWS S3 and Google Cloud’s GCS or vice versa._
+
+## Architecture
+
+COSI is made up of three components:
+   * COSI Controller Manager
+   * COSI Sidecar
+   * COSI Driver
+
+The COSI Controller Manager acts as the main controller that processes changes to COSI API objects. It is responsible for fielding requests for bucket creation, updates, deletion and access management. Only one instance of the controller manager is required for all the object storage providers in the cluster. 
+
+The COSI Sidecar acts as a translator between COSI API requests and vendor specific COSI Drivers. This component uses a standardized gRPC protocol that vendor drivers are expected to satisfy. 
+
+The COSI Driver is the vendor specific component that receives requests from the sidecar and calls the appropriate vendor APIs to create buckets, manage their lifecycle and manage access to them. 
+
+## API
+
+The COSI API is centered around buckets, since they are the unit of abstraction for provisioning object storage. We define three kubernetes APIs aimed at managing them
```

Comment:
```suggestion
The COSI API is centered around buckets, since they are the unit of abstraction for
provisioning object storage. There are three Kubernetes APIs (implemented as
[CustomResourceDefinitions](/docs/concepts/extend-kubernetes/api-extension/custom-resources/))
aimed at managing these:
```